### PR TITLE
feat(Loading): migrate to Tailwind

### PR DESCRIPTION
**Pull request by mainframev**
[FEPL-1674](https://kiwicom.atlassian.net/browse/FEPLT-1674)
 Storybook: https://orbit-mainframev-tw-migrate-loading.surge.sh

### CODE REVIEW

File: packages/orbit-components/src/Loading/index.tsx [hunk 1]
```diff
@@ -1,191 +1,70 @@
...
     return (
-      <StyledLoader>
-        <StyledLoaderCircle />
-        <StyledLoaderCircle />
-        <StyledLoaderCircle />
-      </StyledLoader>
+      <div className="flex items-center justify-center">
+        <div className="animate-loader rounded-circle bg-cloud-dark h-xs w-xs me-[6px]" />
+        <div
+          className="animate-loader rounded-circle bg-cloud-dark h-xs w-xs me-[6px]"
```
Comment (DSil) [inline packages/orbit-components/src/Loading/index.tsx:43]: I see this `animate-loader` class applies `loader 1s 1.25s infinite ease-in-out` but it used to be `loader 1.25s infinite ease-in-out`. This must be changed on the tailwind-preset package.

File: packages/orbit-components/src/Loading/index.tsx [hunk 2]
```diff
@@ -1,191 +1,70 @@
...
   return (
-    <StyledSpinner viewBox="0 0 40 40">
-      <StyledSpinnerCircle cx="50%" cy="50%" r="18" type={type} />
-    </StyledSpinner>
+    <svg viewBox="0 0 40 40" className="orbit-loading-spinner animate-spinner h-xxl w-xxl">
+      <circle
+        cx="50%"
+        cy="50%"
+        fill="transparent"
+        strokeWidth={3}
```
Comment (DSil) [inline packages/orbit-components/src/Loading/index.tsx:60]: Same question as above

File: packages/orbit-components/src/Loading/index.tsx [hunk 3]
```diff
@@ -1,191 +1,70 @@
...
+        viewBox={`0 0 ${customSize} ${customSize}`}
+        className="orbit-loading-spinner animate-spinner h-xxl w-xxl"
+      >
+        <circle
           cx="50%"
           cy="50%"
-          r={customSize ? customSize / 2 - 2 : 18}
-          customSize={customSize}
+          fill="transparent"
+          strokeWidth={3}
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Loading/index.tsx:27]: it won't make any difference, so it's ok

File: packages/orbit-components/src/Loading/index.tsx [hunk 4]
```diff
@@ -200,15 +79,38 @@ const Loading = ({
...
+      {Boolean(children) && !loading ? (
         children
       ) : (
-        <StyledLoading type={type} dataTest={dataTest} id={id}>
+        <div
+          className={cx([
+            "flex",
+            "items-center",
+            "justify-center",
+            "h-[120px]",
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/Loading/index.tsx:90]: it will be a default and if type has it's own height -> it will override that value above, but agree that better to be explicit here and define for each type.

File: packages/orbit-components/.eslintrc.cjs [hunk 5]
```diff
@@ -30,6 +30,10 @@ module.exports = {
       files: ["src/**/*.ts?(x)"],
       rules: {
         "@typescript-eslint/consistent-type-imports": "error",
+        "tailwindcss/no-custom-classname": [
+          1,
+          { whitelist: ["(orbit\\-).*"], config: "./tailwind.config.ts" },
```
Comment (DSil) [inline packages/orbit-components/.eslintrc.cjs:35]: Why this change? I believe that if we want to use custom components in our docs, for example, where we have some, we will face same problem. So why can't we have it global?

File: packages/orbit-components/src/Loading/index.tsx [hunk 6]
```diff
@@ -1,194 +1,76 @@
...
-    margin: 0;
-  }
-`;
-
-StyledLoaderCircle.defaultProps = {
-  theme: defaultTheme,
+const CircleLoader = ({ animationDelay }: { animationDelay?: string }) => {
+  return (
+    <div
+      className="animate-loader rounded-circle bg-cloud-dark h-xs w-xs me-[6px] [&>*:nth-child(3)]:my-0"
```
Comment (DSil) [inline packages/orbit-components/src/Loading/index.tsx:13]:
```suggestion
      className="animate-loader rounded-circle bg-cloud-dark h-xs w-xs me-[6px] [&:nth-child(3)]:m-0"
```

You were targeting descendants with the nth(3) property, but you wanna target the actual element. If it is the third child of its parent, it should have `margin: 0` (and not `margin-y: 0` 😉).

File: packages/orbit-components/.eslintrc.cjs (hunk 5 above)
Comment (mainframev) [reply packages/orbit-components/.eslintrc.cjs:35]: because it needs the reference to tailwind.config.ts, which is not global (it can't find our preset classes without it and eslint throws warnings) 
Screenshot 2023-09-07 at 12 16 27

Comment (mainframev) [reply packages/orbit-components/.eslintrc.cjs:35]: I can try to have it globally and just import the object from our preset

Comment (DSil) [reply packages/orbit-components/.eslintrc.cjs]: I don't think that is the reason. The warning is thrown every time a new class is used for the first time, until CSS is built. So if you see that warning what you can do is build the CSS `yarn components tailwind` and, if needed, restart the ESLint server on VSCode. Those steps have been enough to solve those warnings to me

Comment (mainframev) [reply packages/orbit-components/.eslintrc.cjs]: I thought it would eliminate the need to build the file

Comment (mainframev) [reply packages/orbit-components/.eslintrc.cjs]: yep, we can

**Review by DSil — APPROVED** [review]
